client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -119,6 +120,32 @@
         setNodes(nodes);
     }
 
+    /**
+     * Returns a new {@link RestClientBuilder} to help with {@link RestClient} creation.
+     * Creates a new builder instance and sets the nodes that the client will send requests to.
+     *
+     * @param cloudId a valid elastic cloud cloudId that will route to a cluster
+     */
+    public static RestClientBuilder builder(String cloudId) {
+        // there is an optional first portion of the cloudId that is a human readable string, but it is not used.
+        if (cloudId.contains(":")) {
+            if (cloudId.indexOf(":") == cloudId.length() - 1) {
+                throw new IllegalStateException("cloudId " + cloudId + " is invalid");
+            }
+            cloudId = cloudId.substring(cloudId.indexOf(":") + 1);
+        }
+
+        String decoded = new String(Base64.getDecoder().decode(cloudId));
+        // once decoded the parts are separated by a $ character
+        String[] decodedParts = decoded.split("\\$");
+        if (decodedParts.length != 3) {
+            throw new IllegalStateException("cloudId " + cloudId + " did not contain the correct number of parts");
```

Comment:
This error will be really confusing for users who have no idea that cloudIds contain multiple parts, or that they are even encoded at all. Maybe we can change it to something like:

"clouldId ... is invalid and does not decode to a cluster identifier correctly"

Reply:
well the cloudID is masked for the user. It would be a copy/paste issue since these are generated and displayed in the UI for a user. But i can change it for sure.